### PR TITLE
[fix](scanner) fix load cannot end when set exec_mem_limit

### PR DESCRIPTION
**Pull request by sollhui**
## Proposed changes

When value of exec_mem_limit is too small(like 1 byte), the load cannot be completed.
```
int get_available_thread_slot_num() {
        int thread_slot_num = 0;
        thread_slot_num = (allowed_blocks_num() + _block_per_scanner - 1) / _block_per_scanner;
        thread_slot_num = std::min(thread_slot_num, _max_thread_num - _num_running_scanners);
        return thread_slot_num;
}

int32_t allowed_blocks_num() const {
        int32_t blocks_num = std::min(_free_blocks_capacity,
                                      int32_t((_max_bytes_in_queue + _estimated_block_bytes - 1) /
                                              _estimated_block_bytes));
        return blocks_num;
}
```
When exec_mem_limit = 1, _max_bytes_in_queue = 1/20 * num_parallel_instances = 0，and the value of get_available_thread_slot_num() is 0, which none thread will add block to queue and make sender in vtablet_sink endless loop.

Solving it by set minimum max_bytes_in_blocks_queue threshold to optimize get_available_thread_slot_num() logic.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

File: be/src/vec/exec/scan/scanner_context.cpp [hunk 1]
```diff
@@ -55,7 +55,6 @@ ScannerContext::ScannerContext(doris::RuntimeState* state_, doris::vectorized::V
           _process_status(Status::OK()),
           _batch_size(state_->batch_size()),
           limit(limit_),
-          _max_bytes_in_queue(max_bytes_in_blocks_queue_ * num_parallel_instances),
```
Comment (yiguolei) [inline be/src/vec/exec/scan/scanner_context.cpp:58]: your modification is too complicated, 
just rewrite it to  std::max(max_bytes_in_blocks_queue_, 1024)*num_parallel_instances

and please add comment here to explain why we need do like this.

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by freemandealer — APPROVED** [review]
LGTM

**Pull request merged**